userservice/src/main/java/ro/devdepot/services/implementation/UserServiceImpl.java

```diff
@@ -27,4 +33,42 @@ public LoginResponse login(LoginRequest loginRequest) {
 
         return null;
     }
+
+    @Override
+    public User createUser(User user) {
+        return userRepository.save(user);
+    }
+
+    @Override
+    public User updateUser(User user, Long id) {
+        User existingUser = userRepository.findById(id).orElse(null);
+
+        existingUser.setUsername(user.getUsername());
```

Comment:
user should not be able to update its username and role, only other fields like name, email etc, for password will be another method, also add validation on updated user ( you could create a private method in the service called validateUserDto(User user) and use it everywhere you need to use the same validations